melkor/base/dataset.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+import pandas as pd
+import numpy as np
+from typing import Tuple
+
+
```

Comment:
I get the end goal of this object, but wouldn't it be better to have a catalogue object that knows how read the data, also controls permissions (not relevant right now), and has I/O operations for different formats and sources? because here you still need to manually pass or define the paths in the code and that is neither scalable nor maintainable